src/agent/src/util.rs

```diff
@@ -172,90 +173,58 @@ mod tests {
-            let reader = Cursor::new(d.reader_value.clone());
-            let writer = BufWriter::new();
+    async fn test_interruptable_io_copier_reader(#[case] reader_value: String) {
+        let (tx, rx) = channel(true);
+        let reader = Cursor::new(reader_value.clone());
+        let writer = BufWriter::new();
 
-            // XXX: Pass a copy of the writer to the copier to allow the
-            // result of the write operation to be checked below.
-            let handle = tokio::spawn(interruptable_io_copier(reader, writer.clone(), rx));
+        // XXX: Pass a copy of the writer to the copier to allow the
+        // result of the write operation to be checked below.
+        let handle = tokio::spawn(interruptable_io_copier(reader, writer.clone(), rx));
 
-            // Allow time for the thread to be spawned.
-            tokio::time::sleep(Duration::from_secs(1)).await;
+        // Allow time for the thread to be spawned.
+        tokio::time::sleep(Duration::from_secs(1)).await;
 
-            let timeout = tokio::time::sleep(Duration::from_secs(1));
-            pin!(timeout);
+        let timeout = tokio::time::sleep(Duration::from_secs(1));
+        pin!(timeout);
 
-            // Since the readers only specify a small number of bytes, the
-            // copier will quickly read zero and kill the task, closing the
-            // Receiver.
-            assert!(tx.is_closed(), "{}", msg);
+        // Since the readers only specify a small number of bytes, the
+        // copier will quickly read zero and kill the task, closing the
+        // Receiver.
+        assert!(tx.is_closed());
 
-            let spawn_result: std::result::Result<
-                std::result::Result<u64, std::io::Error>,
-                JoinError,
-            >;
+        let spawn_result: std::result::Result<std::result::Result<u64, std::io::Error>, JoinError>;
 
-            select! {
-                res = handle => spawn_result = res,
-                _ = &mut timeout => panic!("timed out"),
-            }
+        select! {
+            res = handle => spawn_result = res,
+            _ = &mut timeout => panic!("timed out"),
+        }
 
-            assert!(spawn_result.is_ok());
+        assert!(spawn_result.is_ok());
 
-            let result: std::result::Result<u64, std::io::Error> = spawn_result.unwrap();
+        let result: std::result::Result<u64, std::io::Error> = spawn_result.unwrap();
 
-            assert!(result.is_ok());
+        assert!(result.is_ok());
 
-            let byte_count = result.unwrap() as usize;
-            assert_eq!(byte_count, d.reader_value.len(), "{}", msg);
+        let byte_count = result.unwrap() as usize;
+        assert_eq!(byte_count, reader_value.len());
 
-            let value = writer.to_string();
-            assert_eq!(value, d.reader_value, "{}", msg);
-        }
+        let value = writer.to_string();
+        assert_eq!(value, reader_value);
     }
 
+    #[rstest]
```

Comment:
Similar question on here as the other commit, whether the non-modified tests need to be switched to `rstest`